### PR TITLE
docs: replaces json patch bodies with yaml format

### LINKED ISSUE #3180: docs: make kubectl patch commands use yaml instead of json payloads, for legibility
*Description*:

i wonder whether the `kubectl patch` commands used in different guides can be recast using yaml instead of json, to eliminate some of that noise.

example:

https://gateway.envoyproxy.io/v1.0.1/tasks/security/basic-auth/#enable-https

```shell
kubectl patch gateway eg --type=json --patch '[{
   "op": "add",
   "path": "/spec/listeners/-",
   "value": {
      "name": "https",
      "protocol": "HTTPS",
      "port": 443,
      "tls": {
        "mode": "Terminate",
        "certificateRefs": [{
          "kind": "Secret",
          "group": "",
          "name": "example-cert",
        }],
      },
    },
}]'
```

becomes..

```shell
cat <<EOF | kubectl patch gateway eg --type=json --patch-file=/dev/stdin
- op: add
  path: "/spec/listeners/-"
  value:
    name: https
    protocol: HTTPS
    port: 443
    tls:
      mode: Terminate
      certificateRefs:
      - group: ""
        kind: Secret
        name: example-cert
EOF
```

i'm not sure the yaml example above is the canonical way of using kubectl patch with yaml.  i'll look more closely at that if this issue is accepted post-triage.

### PR DESCRIPTION
**Pull request by eitansuez**
**What type of PR is this?**

* docs: replaces kubectl patch with json bodies to yaml, for better legibility

**What this PR does / why we need it**:

improves the docs.

**Which issue(s) this PR fixes**:

Fixes #3180

### CODE REVIEW

File: site/content/en/latest/tasks/traffic/gateway-address.md [hunk 1]
```diff
@@ -21,21 +22,22 @@ The Envoy Gateway deploys Envoy Proxy Service as `LoadBalancer` by default,
...
+'
 ```
 
 Verify the Gateway status:
 
 ```shell
 kubectl get gateway
+```
 
+```console
```
Comment (yuluo-yx) [inline site/content/en/latest/tasks/traffic/gateway-address.md:40]: imo, we can use markdown style.  console doesn't seem to highlight any code in md.

```markdown
NAME   CLASS   ADDRESS   PROGRAMMED   AGE
eg     eg      1.2.3.4   True         14m
```

```console
NAME   CLASS   ADDRESS   PROGRAMMED   AGE
eg     eg      1.2.3.4   True         14m
```

File: site/content/en/v1.0.1/tasks/security/backend-tls.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ Create a certificate and a private key for `www.example.com`:
 
 ```shell
 openssl req -out www.example.com.csr -newkey rsa:2048 -nodes -keyout www.example.com.key -subj "/CN=www.example.com/O=example organization"
-openssl x509 -req -days 365 -CA example.com.crt -CAkey example.com.key -set_serial 0 -in www.example.com.csr -out www.example.com.crt
+openssl x509 -req -days 365 -CA ca.crt -CAkey ca.key -set_serial 0 -in www.example.com.csr -out www.example.com.crt
```
Comment (arkodg) [inline site/content/en/v1.0.1/tasks/security/backend-tls.md:32]: same as above

File: site/content/en/latest/tasks/traffic/gateway-address.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ title: "Gateway Address"
 The Gateway API provides an optional [Addresses][] field through which Envoy Gateway can set addresses for Envoy Proxy Service.
 Depending on the Service Type, the addresses of gateway can be used as:
 
+- [Prerequisites](#prerequisites)
```
Comment (shawnh2) [inline site/content/en/latest/tasks/traffic/gateway-address.md:8]: do we need this in this context?

File: site/content/en/v1.0.1/tasks/security/mutual-tls.md [hunk 4]
```diff
@@ -2,16 +2,16 @@
...
+This task uses a self-signed CA, so it should be used for testing and demonstration purposes only.
 
 ## Prerequisites
 
 - OpenSSL to generate TLS assets.
 
 ## Installation
 
-Follow the steps from the [Quickstart](../../quickstart) to install Envoy Gateway and the example manifest.
+Follow the steps from the [Quickstart](../quickstart) to install Envoy Gateway and the example manifest.
```
Comment (shawnh2) [inline site/content/en/v1.0.1/tasks/security/mutual-tls.md:14]: this link should not be changed, right?

File: site/content/en/v1.0.1/tasks/traffic/gateway-address.md [hunk 5]
```diff
@@ -5,6 +5,7 @@ title: "Gateway Address"
 The Gateway API provides an optional [Addresses][] field through which Envoy Gateway can set addresses for Envoy Proxy Service.
 Depending on the Service Type, the addresses of gateway can be used as:
 
+- [Prerequisites](#prerequisites)
```
Comment (shawnh2) [inline site/content/en/v1.0.1/tasks/traffic/gateway-address.md:8]: ditto

File: site/content/en/v1.0.1/tasks/security/secure-gateways.md [hunk 6]
```diff
@@ -2,16 +2,16 @@
...
+This task uses a self-signed CA, so it should be used for testing and demonstration purposes only.
 
 ## Prerequisites
 
 - OpenSSL to generate TLS assets.
 
 ## Installation
 
-Follow the steps from the [Quickstart](../../quickstart) to install Envoy Gateway and the example manifest.
+Follow the steps from the [Quickstart](../quickstart) to install Envoy Gateway and the example manifest.
```
Comment (shawnh2) [inline site/content/en/v1.0.1/tasks/security/secure-gateways.md:14]: ditto

File: site/content/en/v1.0.1/tasks/security/secure-gateways.md [hunk 7]
```diff
@@ -257,10 +251,10 @@ This section gives a walkthrough to generate RSA and ECDSA derived certificates
 
 ## Prerequisites
 
-Follow the steps from the [Quickstart](../../quickstart) to install Envoy Gateway and the example manifest.
+Follow the steps from the [Quickstart](../quickstart) to install Envoy Gateway and the example manifest.
```
Comment (shawnh2) [inline site/content/en/v1.0.1/tasks/security/secure-gateways.md:254]: ditto

**Review by shawnh2 — COMMENTED** [review]
thanks for the amazing work! just add few comments

File: site/content/en/v1.0.1/tasks/security/secure-gateways.md (hunk 6 above)
Comment (zirain) [reply site/content/en/v1.0.1/tasks/security/secure-gateways.md:14]: Actually, the new one is the more correct one.

**Review by arkodg — APPROVED** [review]
thanks !

**Review by shawnh2 — APPROVED** [review]
lgtm

**Pull request merged**